pkg/noun/events.c

```diff
@@ -181,12 +181,45 @@ u3e_check(c3_c* cap_c)
 static inline c3_i
 _ce_flaw_protect(c3_w pag_w)
 {
+  // NB: if this is not static, sometimes we SIGSEGV in the memcpy,
+  // and I don't know why.
+  //
+  static c3_y con_y[16384];
```

Comment:
The stack is grown via page faults, and a non-static declaration would be allocated on the stack at call-time.

Reply:
Is that a general property of C stacks?  And the reason it makes a difference here is that we're already inside a SIGSEGV handler?

Reply:
It's an OS property; I believe all the major ones do something similar. I've written a buggy SIGSEGV handler before, and had it trigger itself endlessly. Was that what you were seeing, or did it just crash?

Reply:
It crashed with SIGSEGV (ie the king printed that the serf crashed with signal 11).  I went on a wild goose chase for a while assuming the memcpy was SIGSEGV'ing because the source was PROT_NONE, but eventually I narrowed it down to writing anything to the 7608th byte in `con_y`, which is consistent with "your stack was about halfway through this page, and you filled up the rest of the page".  Should have checked whether that was on a page boundary, but most likely it was.

Reply:
The stack is grown via page faults, but those aren't delivered as SIGSEGV's and are handled by the kernel transparently. The reason is probably that signal delivery happens on a seperate stack from normal execution, and I believe most libcs set a fixed sized memory region for that instead of an actual MAP_GROWDOWN page.

Reply:
Having it as a static means you're wasting 16kb of memory all the time, though, instead of it being available to the system outside the sigsegv handler. It could be replaced with a seperate mmap'd and cleaned up temporary memory region instead.